src/vivarium/framework/engine.py

```diff
@@ -46,13 +46,39 @@
 
 
 class SimulationContext:
+
+    _created_simulation_contexts: Set[str] = set()
+
+    @staticmethod
+    def _get_context_name(sim_name: Union[str, None]) -> str:
+        # Not threadsafe, but I don't see why we'd need to be.
```

Comment:
You should add docstring making it clear that there is a side-effect here (i.e. adding the simulation to the set of simulation context names